src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -155,6 +159,38 @@ public void close() {
     }
   }
 
+  /**
+   * Determine the acceptable means of authentication for a given DRS ID, including the passport
+   * issuers when supported.
+   *
+   * @param drsObjectId the object ID for which to look up authorizations
+   * @return the `DrsAuthorizations` for this ID
+   * @throws IllegalArgumentException if there is an issue with the object id
+   * @throws SnapshotNotFoundException if the snapshot for the DRS object cannot be found
+   * @throws TooManyRequestsException if there are too many concurrent DRS lookup requests
+   */
+  public DRSAuthorizations lookupAuthorizationsByDrsId(String drsObjectId) {
+    try (DrsRequestResource r = new DrsRequestResource()) {
+      DRSAuthorizations auths = new DRSAuthorizations();
+
+      auths.addSupportedTypesItem(DRSAuthorizations.SupportedTypesEnum.BEARERAUTH);
+      // TODO: add to bearer_auth_issuers, ask Muscles.
+
+      SnapshotCacheResult snapshot = lookupSnapshotForDRSObject(drsObjectId);
+      SnapshotSummaryModel snapshotSummary = getSnapshotSummary(snapshot.id);
+
+      String phsId = snapshotSummary.getPhsId();
+      String consentCode = snapshotSummary.getConsentCode();
+
+      if (phsId != null && consentCode != null) {
```

Comment:
It makes sense to let a SnapshotSummary object make the decision about supporting passport auth by inspecting it's internal state, rather than having the DrsService pull fields out and look at them itself.

Reply:
Does it? I think it's specifically a thing that (at least now), only Drs cares about

Reply:
Depends on how "hardcore OO" we want to be. 😅

Reply:
I'd be happy to "parking lot" this at stand up!  I'll say that it wasn't clear to me how one alters the `SnapshotSummary` from the `SnapshotSummaryModel` (which is a generated file).

Reply:
hahaha.  I think we want to move PhsId and consent code to the cache anyways here so I think this comment might not even apply anymore